### PR TITLE
Remove many instances of catching exceptions

### PR DESCRIPTION
**Pull request by collindutter**
- [x] I have read and agree to the contributing guidelines for [submitting new pull requests](https://github.com/griptape-ai/griptape?tab=readme-ov-file#submitting-pull-requests).

### Changed
- **BREAKING**: Drivers, Loaders, and Engines will now raises exceptions rather than returning `ErrorArtifact`s.

Note that this change was not made to Tools, Tasks, or Structures for the following reasons:
- Tools: Tool output always goes to the LLM, therefore if a Tool errors out we want to convey that to the LLM so that it can course correct. A Tool does not need to return `ErrorArtifact`s on its own, it can raise an exception which will be [caught and converted](https://github.com/griptape-ai/griptape/blob/be18083838eb5dc3c5db780ec43eb966b4e0ac52/griptape/tools/base_tool.py#L120-L129) in BaseTool.
- Tasks & Structures: it is useful for Tasks and Structures to have an output value even if they errored. If [fail_fast](https://github.com/griptape-ai/griptape/blob/be18083838eb5dc3c5db780ec43eb966b4e0ac52/griptape/structures/structure.py#L37) is disabled, the Structure will continue running and you can inspect the output of all Tasks (successful or not) afterwards.

## Issue ticket number and link
NA

----
📚 Documentation preview 📚: https://griptape--1101.org.readthedocs.build//1101/

### CODE REVIEW

File: griptape/tasks/code_execution_task.py [hunk 1]
```diff
@@ -1,19 +1,18 @@
...
 @define
 class CodeExecutionTask(BaseTextInputTask):
     run_fn: Callable[[CodeExecutionTask], BaseArtifact] = field(kw_only=True)
 
     def run(self) -> BaseArtifact:
-        try:
-            return self.run_fn(self)
-        except Exception as e:
-            return ErrorArtifact(f"error during Code Execution Task: {e}")
+        return self.run_fn(self)
```
Comment (vachillo) [inline griptape/tasks/code_execution_task.py:18]: other tasks dont do this do they?

File: griptape/tools/file_manager/tool.py [hunk 2]
```diff
@@ -93,9 +93,16 @@ def save_memory_artifacts_to_disk(self, params: dict) -> ErrorArtifact | InfoArt
...
+            try:
+                self.file_manager_driver.save_file(os.path.join(dir_name, formatted_file_name), artifact.value)
+            except FileNotFoundError:
+                return ErrorArtifact("Path not found")
+            except IsADirectoryError:
+                return ErrorArtifact("Path is a directory")
+            except NotADirectoryError:
+                return ErrorArtifact("Not a directory")
+            except Exception as e:
+                return ErrorArtifact(f"Failed to load file: {str(e)}")
```
Comment (vachillo) [inline griptape/tools/file_manager/tool.py:105]: should this just be the one `except` statement?

**Review by vachillo — APPROVED** [review]

File: griptape/tasks/code_execution_task.py (hunk 1 above)
Comment (collindutter) [reply griptape/tasks/code_execution_task.py:18]: They do not which is why I removed it. `BaseTask` [will catch this already](https://github.com/griptape-ai/griptape/blob/f2b529ab361c33b2a26a4510f78f85181b96af4f/griptape/tasks/base_task.py#L163-L178).

File: griptape/tools/file_manager/tool.py (hunk 2 above)
Comment (collindutter) [reply griptape/tools/file_manager/tool.py:105]: I don't think so. We want to convey each of the exceptions to the LLM in a slightly different way. These are copied directly from the Driver that previously threw them.

**Pull request merged**